scripts/dts/gen_defines.py

```diff
@@ -579,6 +583,46 @@ def write_clocks(node):
                 controller.props["clock-frequency"].val)
 
 
+def write_dmas(node):
```

Comment:
Would be a good idea to reuse the generic `write_phandle_val_list()` here for as much stuff as possible. Hopefully, the identifiers can fit into the same scheme used for other phandle arrays (`pwms`, etc.).

Try just calling just `write_phandle_val_list()` first, see what `#define`s you miss, and output those manually afterwards.

Reply:
write_phandle_val_list() should be only outputting 'channel' 'slot' 'feature' and 'channel_configuration' here, which are declared in cells. I still have to decode 'periph_inc', 'mem_inc', 'periph_inc_fixed', 'priority' and 'fifo_threshold' here, right?

Reply:
We have to get these vars from dts but for complying https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/plain/Bindings/dma/stm32-dma.txt, I can only use 'channel', 'slot', 'channel_config' and 'features' in cells. write_phandle_val_list() will only save one line of code here.

Reply:
Still better to use `write_phandle_val_list()` it for whatever it can be used for I think (and outputting whatever else needs to be output after calling it).

It also forces things like naming consistency, which is nice (`DMAS_CONTROLLER_...` instead of `DMA_CONTROLLER_...`, etc.) We have some inconsistency there for clocks already, which we want to get rid of.